model/src/main/kotlin/DependencyGraphNavigator.kt

```diff
@@ -45,9 +45,17 @@ class DependencyGraphNavigator(
     override fun scopeNames(project: Project): Set<String> = project.scopeNames.orEmpty()
 
     override fun directDependencies(project: Project, scopeName: String): Sequence<DependencyNode> {
-        val graph = graphForManager(project.id.type)
+        // TODO: Relax the assumption that package manager name start with the name of the type of project
+        //       they manage, for example that "GradleInspector" manages "Gradle" projects.
+        val managers = graphs.keys.filter { it.startsWith(project.id.type) }
```

Comment:
Is  the trick here that `GradleInspector` startsWith `Gradle` ?

Would it be an alternative to extend package manager API by providing the projectTypes it may return ?

Reply:
> Is the trick here that `GradleInspector` startsWith `Gradle` ?

Yes, that's what the code comment is trying to explain.

> Would it be an alternative to extend package manager API by providing the projectTypes it may return ?

I've considered that as it certainly would be cleaner, but it would be a much larger refactoring (that I currently cannot afford spend time on). One of the problems is that the dependency graph logic currently only has access to the factories, not the package manager instances themselves, so just adding a `projectType` (and consistently also a `packageType` property) there would not help.

Ideally those two new properties would be statically available, which is something we could probably look into again after migrating package managers to the new plugin API with its `PluginDescriptor`s.

Reply:
Currently the new plugin API is generic for all plugin types, but I guess we could find a way to make this possible.